### PR TITLE
[Web] Compilation with the LMLayer

### PR DESCRIPTION
**Pull request by jahorton**
This is the first step in linking the common LMLayer code into KMW; while not utilized at this time, KMW will compile with full LMLayer code embedded within it.  Actually _using_ and _communicating_ with the LMLayer will come later.

Unfortunately, due to current TS limitations, the LMLayer's WebWorker code does not come with source mapping - see https://github.com/Microsoft/TypeScript/issues/13944.  The outer shell is still source mapped, as the LMLayer TS code is directly linked.  Note that KMW 12.0 will now technically target ES6, so we may need to make extra effort to ensure proper backward compatibility with older browsers.  (Let's not go crazy with it, but some polyfill use should be fine.)

Note that part of this work also moves the LMLayer's public object (the outer shell) into an appropriate namespace, resulting in lots of whitespace changes.  [`?w=1`](https://github.com/keymanapp/keyman/pull/1602/files?w=1) may prove helpful in filtering out the whitespace-only aspects.

### CODE REVIEW

**Comment by keyman-server** [comment]
[OSK renders for Windows](https://build.palaso.org/repository/download/KeymanDesktop_TestPrRenderOnScreenKeyboards/141577:id/kvk/index.html)

**Comment by mcdurdin** [comment]
> but issues arise due to slight differences in their environmental setup and expectations

Is there a good reason to have different environmental setup and expectations? Given the LMLayer is intended to integrate with KMW aren't we better off with a clean compatible setup?

I'd be open to moving the LMLayer stuff into the web folder if that simplifies the build.

**Comment by jahorton** [comment]
It probably wouldn't be an issue if we shifted KMW to compile targeting ES6 instead of ES5, though I'm not comfortable doing that without a little discussion.  Obviously, we can't expect to maintain our browser backward-compatibility if we wholesale target ES6; targeting ES5 helps prevent many of those issues.  As long as we're careful to double-check compatibility tables when using new features, we'd probably be fine... would this change be acceptable?

**Comment by jahorton** [comment]
I suppose either suggested approach would at least give us source mapping for the outer portion of the LMLayer, though it's doubtful that we can get source mapping in KMW for the WebWorker portion.

**Comment by eddieantonio** [comment]
Hey @jahorton! I've been busy with NRC advisory committee stuff. I'll do a proper review of this PR by my Friday (I think that's Saturday for you).

File: web/unit_tests/cases/engine_chirality.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
 var assert = chai.assert;
 
 describe('Engine - Chirality', function() {
-  before(function(done) {
-    this.timeout(kmwconfig.timeouts.scriptLoad);
+  this.timeout(kmwconfig.timeouts.scriptLoad);
 
+  before(function(done) {
```
Comment (jahorton) [inline web/unit_tests/cases/engine_chirality.js]: The placement of this timeout's been causing some CI instability.  This should help with that, hopefully.

**Comment by jahorton** [comment]
Okay, as of the latest commit, we're having some issues with our BrowserStack CI - apparently a few of the Mac tests weren't able to run, but otherwise no tests failed.

**Comment by eddieantonio** [comment]
> Okay, as of the latest commit, we're having some issues with our BrowserStack CI - apparently a few of the Mac tests weren't able to run, but otherwise no tests failed.

Is this simply due to BrowserStack instability?

**Comment by jahorton** [comment]
It's something that's happened before a few times - usually it either lasts a couple of days or it means that they've changed which test platform targets are available.

**Comment by jahorton** [comment]
Huh.  BrowserStack's macOS/Safari targets are still falling over, though they're still listed as valid targets on their site.  Hopefully they sort the configuration out soon.

**Pull request merged**